### DISCUSSION TITLE
Modal / GUI window in a single page application.

### DISCUSSION THREAD

**Discussion by aykaramba** [rabbibotton/clog#342] [General]
I am working on a single page application and everything is working out great.  I was looking at the clog-gui examples and w3.css modal css options to open up pseudo pop-ups to display "modal" information in the spa style application.

From what I gather, clog-gui is basically for creating gui style applications and you work with the entire modal metaphor presented in that project template.  This probably means that if I want a modal style popup to display some statistics for an item, I probably have to look at w3.css modal popups and use some javascript / other customization to make the pop-up dragable as well.

Am I understanding my options correctly?  Or are there any options for re-purposing the modal windows found in clog-gui that might allow for dragable popups to be displayed?

Thanks.

**Comment by rabbibotton** [comment]
There is support for modal popups etc in clog-gui not sure what you mean? You can create any type of window and then do window-make-modal and window-end-modal, also see for example input-dialog and the :modal argument also worth seeing the :time-out option for combing the ability to block till dialog done and time outs etc

**Comment by aykaramba** [comment]
Oh, i don't have to use the clog-gui template / desktop metaphor?  Okay cool, let me look into that a bit more, I couldn't figure that out.  Thanks.

**Comment by rabbibotton** [comment]
dont create a menu bar, etc

**Comment by aykaramba** [comment]
For anyone wondering:

1) The examples are in 22-tutorial.lisp

2) You need to initialize the following on-new-window or similar:

```
  (clog-gui-initialize body) ;; initialized clog-gui
  (enable-clog-popup) ;; allows browser pop-ups
```

3) You can do something like this:

```
(set-on-click btn-window-notifications
    (lambda (obj)
                     (let* ((win (create-gui-window body :title "Notifications" :client-movement t :width 400 :height 400 :keep-on-top t :title-class "w3-center w3-black")))
                    (create-div win :content "hello!"))))
```

As mentioned earlier, no need to load up the menu components if you don't want to.
